pkg/app/piped/cloudprovider/kubernetes/state.go

```diff
@@ -534,3 +564,209 @@ func determineServiceAccountHealth(obj *unstructured.Unstructured) (status model
 	return
 }
+
+// This function based on [istio project](https://github.com/istio/istio/blob/8ce0defcc905873dadf3fa1d8c3f3629cd39895c/pkg/config/validation/validation.go#L1930-L2041).
+// See the file headers for detail information.
+func determineVirtualService(obj *unstructured.Unstructured) (status model.KubernetesResourceState_HealthStatus, desc string) {
+	vs := &istio.VirtualService{}
+	err := scheme.Scheme.Convert(obj, vs, nil)
+	if err != nil {
+		status = model.KubernetesResourceState_OTHER
+		desc = fmt.Sprintf("Unexpected error while calculating: unable to convert %T to %T: %v", obj, vs, err)
+		return
+	}
+	if len(vs.Hosts) == 0 {
+		desc = "Virtual service must have at least one host"
+		return
+	}
+
+	if len(vs.Http) == 0 && len(vs.Tcp) == 0 && len(vs.Tls) == 0 {
+		desc = "Http, tcp or tls must be provided in virtual service"
+		return
+	}
+	for _, httpRoute := range vs.Http {
+		if httpRoute == nil {
+			desc = "Http route may not be null"
+			continue
+		}
+	}
+	return
+}
+
+// This function based on [istio project](https://github.com/istio/istio/blob/8ce0defcc905873dadf3fa1d8c3f3629cd39895c/pkg/config/validation/validation.go#L442-L484).
+// See the file headers for detail information.
+func determineGateway(obj *unstructured.Unstructured) (status model.KubernetesResourceState_HealthStatus, desc string) {
+	g := &istio.Gateway{}
+	err := scheme.Scheme.Convert(obj, g, nil)
+	if err != nil {
+		status = model.KubernetesResourceState_OTHER
+		desc = fmt.Sprintf("Unexpected error while calculating: unable to convert %T to %T: %v", obj, g, err)
+		return
+	}
+
+	if len(g.Servers) == 0 {
+		status = model.KubernetesResourceState_OTHER
+		desc = "Gateway must have at least one server"
+		return
+	}
+
+	portNames := make(map[string]bool)
+	for _, s := range g.Servers {
+		if s == nil {
+			desc = "Server may not be nil"
+			return
+		}
+		if s.Port != nil {
+			if portNames[s.Port.Name] {
+				desc = fmt.Sprintf("Port names in servers must be unique: duplicate name: %s", s.Port.Name)
+				return
+			}
+			portNames[s.Port.Name] = true
+			switch strings.ToLower(s.Port.Protocol) {
+			case "http", "http2", "http-proxy", "grpc", "grpc-web":
+				if s.GetTls().GetHttpsRedirect() {
+					desc = fmt.Sprintf("Tls.httpsRedirect should only be used with http servers: %T", s.Tls)
+					return
+				}
+			}
+		}
+	}
+	return
+}
+
+// This function based on [istio project](https://github.com/istio/istio/blob/8ce0defcc905873dadf3fa1d8c3f3629cd39895c/pkg/config/validation/validation.go#L611-L653).
+// See the file headers for detail information.
+func determineDestinationRule(obj *unstructured.Unstructured) (status model.KubernetesResourceState_HealthStatus, desc string) {
+	dr := &istio.DestinationRule{}
+	err := scheme.Scheme.Convert(obj, dr, nil)
+	if err != nil {
+		status = model.KubernetesResourceState_OTHER
+		desc = fmt.Sprintf("Unexpected error while calculating: unable to convert %T to %T: %v", obj, dr, err)
+		return
+	}
+
```

Comment:
There is a validation here  as follows:
https://github.com/istio/istio/blob/master/pkg/config/validation/validation.go#L624-L643

However, I think we don't need it because `features.EnableDestinationRuleInheritance` is `false` as an environment variable, and I think we don't change the environment variable in istio.